lib/googleauth/user_refresh.rb

```diff
@@ -39,21 +39,34 @@ class UserRefreshCredentials < Signet::OAuth2::Client
       attr_reader :project_id
       attr_reader :quota_project_id
 
+      # The type name for this credential.
+      CREDENTIAL_TYPE_NAME = "authorized_user".freeze
+
       # Create a UserRefreshCredentials.
       #
       # @param json_key_io [IO] An IO object containing the JSON key
       # @param scope [string|array|nil] the scope(s) to access
       def self.make_creds options = {}
         json_key_io, scope = options.values_at :json_key_io, :scope
-        user_creds = read_json_key json_key_io if json_key_io
-        user_creds ||= {
-          "client_id"     => ENV[CredentialsLoader::CLIENT_ID_VAR],
-          "client_secret" => ENV[CredentialsLoader::CLIENT_SECRET_VAR],
-          "refresh_token" => ENV[CredentialsLoader::REFRESH_TOKEN_VAR],
-          "project_id"    => ENV[CredentialsLoader::PROJECT_ID_VAR],
-          "quota_project_id" => nil,
-          "universe_domain" => nil
-        }
+        if json_key_io
+          json_key = MultiJson.load json_key_io.read
```

Comment:
You're getting rubocop dings for method too long here.
I would recommend taking the verification you've added (in 3 classes) and see if it can be folded in `read_json` method.